File: RAI/Clase16/Java/AbstractClasses/Animal.java
//package RAI.Clase16.Java.AbstractClasses;

public abstract class Animal {
    protected String nombre;
    protected String especie;

    public Animal(String nombre, String especie){
        this.nombre = nombre;
        this.especie = especie;
    }

    @Override
    public String toString(){
        return "Animal " + this.nombre + " de la especie " + this.especie;
    }

    public abstract void haceSonido();//es una declaracion de metodo que ahora es abstracto
}

File: RAI/Clase16/Python/AbstractClasses/Animal.py
from abc import ABC, abstractmethod
# IMportando el comportamiento abstracto desde una metaclase ABC

# La herencia con ABC provoca el comportamiento abstracto de la clase Animal
class Animal(ABC):

    # se pueden definir los atributos de la siguiente manera
    nombre:str
    especie:str
    """
    def __init__(self, nombre, especie):
        self.nombre = nombre
        self.especie = especie
    """

    def __str__(self) -> str:
        return "Animal " + self.nombre + " de especie " + self.especie

    @abstractmethod
    def hace_sonido(self):
        pass